src/main/java/com/twentyonec/ItemsLogger/ItemsLogger.java

```diff
@@ -65,4 +83,28 @@ public void debugMessage(final String message) {
 		if (config.getDebug())
 			plugin.getLogger().log(Level.INFO, "[DEBUG]: " + message);
 	}
+
+	private void registerEvents() {
+		debugMessage("Registering events.");
+		if (config.getDeath()) {
+			debugMessage("Registering Death Event");
+			this.getServer().getPluginManager().registerEvents(new DeathSave(), this);
+		}
```

Comment:
Why are you not registering all events on startup?

```suggestion
debugMessage("Registering Death Event");

this.getServer().getPluginManager().registerEvents(new DeathSave(), this);
```

And then somewhere in the death event class:

```java
public void onDeath(PlayerDeathEvent event) {
    if (config.getDeath) {
        // Handle death event
    }
}
```

Same for all other events.